01.multithreading/MultiThreading.Task6.Continuation/Program.cs

```diff
@@ -22,8 +24,53 @@ static void Main(string[] args)
             Console.WriteLine("Demonstrate the work of the each case with console utility.");
             Console.WriteLine();
 
-            // feel free to add your code
+            var task1 = Task.Run(() => {
+                Console.WriteLine("Run task for case 1");
+            });
 
+            task1.ContinueWith((antecedent) =>
+            {
+                Console.WriteLine("Continuation task is executing regardless of the result of the parent task");
+            });
+
+            var task2 = Task.Run(() =>
+            {
+                Console.WriteLine("Run task for case 2");
+                throw new Exception();
+            });
+
+            task2.ContinueWith((antecedent) =>
+            {
+                Console.WriteLine("Continuation task is executing when the parent task was completed without success");
+            }, TaskContinuationOptions.OnlyOnFaulted);
+
+            var task3 = Task.Run(() =>
+            {
+                Console.WriteLine("Run task for case 3");
+                throw new Exception();
+            });
+
+            task3.ContinueWith((antecedent) =>
+            {
+                Console.WriteLine("Continuation task is executing when the parent task failed and parent task thread should be reused for continuation");
+            }, TaskContinuationOptions.AttachedToParent);
```

Comment:
AttachToParent not suitable here. Review documentation.
To check try to set Thread.CurrentThread.Name in parent task and read it child task.